cli/test/lib/tasks/dependency_spec.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * as of Webpack 5, dependencies that are polyfilled through the Provide plugin must be defined inside the CLI
+ * in order to guarantee there is a version of the dependency accessible by the cypress CLI, either in the cypress directory
+ * or the root of their project. Currently, these two dependencies are 'buffer' and 'process'
+ */
+describe('dependencies', () => {
```

Comment:
I have some questions about this. If webpack 4 polyfilled all core Node.js libs out of the box, does that mean you could (and can) 

- install cypress 12
- write a spec that imports `fs`
- have it run successfully (eg, `fs` won't error out)

I haven't tried this yet, but worth considering. I know we don't do this in our code base, but I'm sure some code bases rely on those polyfills (especially `crypto`).

I wonder if we need to polyfill ALL the core modules to maintain backwards compatibility.

Reply:
I think it depends. `fs` is probably a bad example here since we intentionally leave it empty in the `webpack-batteries-included-preprocessor` and don't expect that built in to be available. But for something like `crypto` for your example, you could do all those steps and it will not error out since we install the polyfill. I think you talked about it in another spot in this PR, but [this comment explains](https://github.com/cypress-io/cypress/pull/27438#discussion_r1281259701) it for anyone else reading